ql/src/java/org/apache/hadoop/hive/ql/exec/tez/DagUtils.java

```diff
@@ -265,11 +279,70 @@ public URI apply(Path path) {
         }
         dag.addURIsForCredentials(uris);
       }
+      getKafkaCredentials((MapWork)work, dag, conf);
     }
-
     getCredentialsForFileSinks(work, dag);
   }
 
+  private void getKafkaCredentials(MapWork work, DAG dag, JobConf conf) {
+    Token<?> tokenCheck = dag.getCredentials().getToken(KAFKA_DELEGATION_TOKEN_KEY);
+    if (tokenCheck != null) {
+      LOG.debug("Kafka credentials already added, skipping...");
+      return;
+    }
+    LOG.info("Getting kafka credentials for mapwork: " + work.getName());
+
+    String kafkaBrokers = null;
+    Map<String, PartitionDesc> partitions = work.getAliasToPartnInfo();
```

Comment:
This is iterating over all partition objects in plan even when kafka is not used. This gets expensive when there are large number of partition objects. Is it possible to do a quick check to see if kafka is used before iterating over full list of parttions?

Reply:
good point, I was thinking about the same, this can become an expensive loop and 100% useless for users not using kafka, let me look a short-circuit way

Reply:
@ashutoshc : what do you think about this? https://github.com/apache/hive/pull/1379/commits/edc4ad440af4e234b731104bbcb9837cbdf43a19#diff-d7b5b051769b68ed5dd602cc30744439R303-R306
(tested on cluster)